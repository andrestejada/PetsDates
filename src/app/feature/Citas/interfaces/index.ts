export interface Cita{
    nombrePropietario:string;
    nombreMascota:string;
    tipoServicio:string;
    tarifa:number;
    fechaHora:string;
    observaciones:string;
}
export interface Citas extends Cita{
    id: number;
}
