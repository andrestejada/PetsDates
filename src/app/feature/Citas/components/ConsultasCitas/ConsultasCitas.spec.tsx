import '@testing-library/jest-dom';
import { Citas } from '../../interfaces/index';
import ConsultarCitas from './index';
import { Provider } from 'react-redux';
import React from 'react';
import createMockStore from 'redux-mock-store';
import { mount} from 'enzyme';
import thunk from 'redux-thunk';


const middlewares = [thunk]; // add your middlewares like `redux-thunk`
const mockStore = createMockStore(middlewares);
interface State{
    dates:{
        allDates:Citas[]
    }
}
const initialState:State={
    dates:{
        allDates:[]
    }
};

let store = mockStore(initialState);

describe('testing <ConsultasCitas/>', () => {
    let componentWrapper = mount(
        <Provider store={store} >
            <ConsultarCitas />
        </Provider>
    );

    beforeEach(()=>{
        componentWrapper = mount(
            <Provider store={store} >
                <ConsultarCitas />
            </Provider>
        );
        store = mockStore(initialState);
        jest.clearAllMocks();    
    });

    it('Comprobar que el componente se renderize correctamente ', () => {
        expect(componentWrapper).toMatchSnapshot();
        expect(componentWrapper.find('ContenedorCitas').exists()).toBe(true);
    });

   
});
