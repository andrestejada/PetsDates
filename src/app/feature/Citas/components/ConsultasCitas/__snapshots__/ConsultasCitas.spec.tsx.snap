// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <ConsultasCitas/> Comprobar que el componente se renderize correctamente  1`] = `
<div
  className="sc-AxiKw kBpZeU"
/>
`;
